AlfrescoApp/View Controllers/FileFolder View Controller/BaseFileFolderCollectionViewController.m

```diff
@@ -106,29 +118,496 @@ - (void)updateSearchResultsForSearchController:(UISearchController *)searchContr
+                {
+                    NSIndexPath *indexPath = [self.collectionView indexPathForItemAtPoint:startingPoint];
+                    if(indexPath && indexPath.item < self.dataSource.numberOfNodesInCollection)
+                    {
+                        self.initialCellForSwipeToDelete = indexPath;
+                    }
+                }
+            }
+            else if (gestureRecognizer.state == UIGestureRecognizerStateChanged)
+            {
+                if(self.initialCellForSwipeToDelete)
+                {
+                    CGPoint translation = [gestureRecognizer translationInView:self.view];
+                    if (translation.x < 0)
+                    {
+                        self.shouldShowOrHideDelete = (translation.x * -1) > self.cellActionViewWidth / 2;
+                    }
+                    else
+                    {
+                        self.shouldShowOrHideDelete = translation.x > self.cellActionViewWidth / 2;
+                    }
+                    
+                    FileFolderCollectionViewCell *cell = (FileFolderCollectionViewCell *)[self.collectionView cellForItemAtIndexPath:self.initialCellForSwipeToDelete];
+                    [cell revealActionViewWithAmount:translation.x];
+                }
+            }
+            else if (gestureRecognizer.state == UIGestureRecognizerStateEnded)
+            {
+                if(self.initialCellForSwipeToDelete)
+                {
+                    if(self.shouldShowOrHideDelete)
+                    {
+                        if(properLayout.selectedIndexPathForSwipeToDelete)
+                        {
+                            properLayout.selectedIndexPathForSwipeToDelete = nil;
+                        }
+                        else
+                        {
+                            properLayout.selectedIndexPathForSwipeToDelete = self.initialCellForSwipeToDelete;
+                        }
+                    }
+                    else
+                    {
+                        FileFolderCollectionViewCell *cell = (FileFolderCollectionViewCell *)[self.collectionView cellForItemAtIndexPath:self.initialCellForSwipeToDelete];
+                        [cell resetView];
+                        properLayout.selectedIndexPathForSwipeToDelete = nil;
+                    }
+                }
+            }
+        }
+    }
+}
+
+- (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldReceiveTouch:(UITouch *)touch
+{
+    if(gestureRecognizer == self.tapToDismissDeleteAction)
+    {
+        if([self.collectionView.collectionViewLayout isKindOfClass:[BaseCollectionViewFlowLayout class]])
+        {
+            BaseCollectionViewFlowLayout *properLayout = (BaseCollectionViewFlowLayout *)self.collectionView.collectionViewLayout;
+            if((properLayout.selectedIndexPathForSwipeToDelete != nil) && (!self.editing))
+            {
+                return YES;
```

Comment:
It's a minor point, but it would be nice to settle on a code style for inline return values. Compare the style used here `return YES;` with that in the method below `shouldBegin = YES; ... return shouldBegin;`